pkg/volume/util/operationexecutor/operation_executor.go

```diff
@@ -216,6 +220,9 @@ type ActualStateOfWorldAttacherUpdater interface {
 	// volumes.  See issue 29695.
 	MarkVolumeAsAttached(volumeName v1.UniqueVolumeName, volumeSpec *volume.Spec, nodeName types.NodeName, devicePath string) error
 
+	// MarkAsAttachedWithSize is same as MarkVolumeAsAttached except we also record pvc size in actualStateOfWorld
+	MarkAsAttachedWithSize(volumeName v1.UniqueVolumeName, volumeSpec *volume.Spec, nodeName types.NodeName, devicePath string, claimSize resource.Quantity) error
```

Comment:
Move this to `type ActualStateOfWorld interface` in `kubernetes/kubernetes/pkg/kubelet/volumemanager/cache/actual_state_of_world.go`. The methods in `ActualStateOfWorldAttacherUpdater` are common for both A/D controller and kubelet volume manager. Since this only applies to the later we don't need to put it in the common interface.

Reply:
We can't move this function into `ActualStateOfWorld` interface of `volumemanager/cache` because - it will cause cyclic dependency.

Reply:
see - cache.ActualStateWorld already depends on `operationexecutor` package, now if we say operationexecutor will depend on `cache.ActualStateWorld` it will cause cyclic dependency.